editor/import/3d/resource_importer_scene.h

```diff
@@ -190,6 +190,17 @@ class ResourceImporterScene : public ResourceImporter {
 		MESH_OVERRIDE_DISABLE,
 	};
 
+	enum PhysicsMode {
+		PHYSICS_DEFAULT,
+		PHYSICS_ENABLE,
+		PHYSICS_DISABLE,
+	};
+
+	enum CollisionMode {
+		COLLISION_DISABLED,
+		COLLISION_STATIC,
+	};
```

Comment:
Why not the other motion types? Also, why doesn't this also have "default"?

Reply:
I think the other types could make sense when Godot finally gets support for indirect collision shape children of rigid bodies.

Basically, I think it is a bit unusual to have an option to automatically convert all mesh instances in the scene to rigid bodies or other types, once per mesh (the scene is not one object, but instead exactly one physical object per mesh). This is very awkward and probably not what the user wants.

I think the user might want to import the whole scene as a single rigid body, but this is where we get the problem with indirect children.

For static bodies, we don't have this problem because you can safely create as many or as few StaticBody3D nodes in the scene without impacting the physical nature.

Reply:
See also https://github.com/godotengine/godot-proposals/issues/542 and https://github.com/godotengine/godot/pull/106048

Honestly, I'd really like to get these merged in first. Having the importer generate CollisionShape3D nodes would be a much better workflow for most cases. so I'd like to see that be both available and given priority, and the scene import code can work around that. So I'd like to consolidate the plan for this PR together with https://github.com/godotengine/godot/pull/106048, but that requires merging https://github.com/godotengine/godot/pull/77937 first, which is still awaiting approval.